flytekit/core/workflow.py

```diff
@@ -504,8 +505,11 @@ def get_input_values(input_value):
                 else:
                     return [input_value]
 
-            # Every time an entity is added, mark it as used.
-            for input_value in get_input_values(kwargs):
+            # Every time an entity is added, mark it as used. The above function though will gather all the input
+            # values but we're only interested in the ones that are Promises so let's filter for those.
+            # There's probably a way to clean this up, maybe key off of the name instead of value?
+            all_input_values = get_input_values(kwargs)
+            for input_value in filter(lambda x: isinstance(x, Promise), all_input_values):
```

Comment:
for non promises we need to add a binding right? is that what self._unbound_inputs means?

Reply:
i don't think so.  let me add a comment to clarify